src/crypto/holder.rs

```diff
@@ -0,0 +1,49 @@
+use once_cell::sync::OnceCell;
+use failure::Fail;
+use super::Cryptographer;
+
+static CRYPTOGRAPHER: OnceCell<&'static dyn Cryptographer> = OnceCell::INIT;
+
+#[derive(Debug, Fail)]
+#[fail(display = "Cryptographer already initialized")]
+pub struct SetCryptographerError(());
```

Comment:
I would think this could just be a panic?  It's definitely a programming error if this occurs, right? It looks like the idea is that the error can be ignored in `autoinit_crypto`, but that seems wrong, too -- if the crate is built with one of the `use_` features, then there should be a panic when the `autoinit_crypto` for that feature is called after `set_cryptographer` is called.  In fact, in the case that either feature is set it probably makes sense to hide the `set_cryptographer` functions.

Reply:
I more or less based this bit on the API for setting the global logger in the `log` crate.

More generally, while it is a programming error:

- There's no API exposed to determine if this function has been called, and I could imagine situations where it's tricky for to be certain about it.
- I'd also prefer to push panics into the consumers, so it's obvious to them that this panics on failure.